### PR TITLE
chore: Migrate CI to GitHub Actions

### PR DESCRIPTION
**Pull request by sawood14012**
added/modified scripts to enable GitHub actions

### CODE REVIEW

File: qa/actions-run-tests.sh [hunk 1]
```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
```
Comment (arajkumar) [inline qa/actions-run-tests.sh:1]: Why not existing qa/runtests.sh?

Comment (sawood14012) [reply qa/actions-run-tests.sh:1]: it has a lot of excess stuff like setting terminal colors , preparing virtual env  etc . added this just to differentiate

Comment (sawood14012) [reply qa/actions-run-tests.sh:1]: also have removed virtual env

Comment (arajkumar) [reply qa/actions-run-tests.sh:1]: Can you fix this review comment?

File: .github/workflows/ci-build.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    - name: Set up Python 3.6
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.6'
+    - name: Run Unit Tests
+      run: bash qa/actions-run-tests.sh
+    - name: Upload coverage to Codecov
+      uses: codecov/codecov-action@v1
+      with:
+        token: d6bd6983-0bad-4eed-b8e3-9fd1d5199257
```
Comment (arajkumar) [inline .github/workflows/ci-build.yml:41]: Read from env variable similar to backbone

**Review by arajkumar — CHANGES_REQUESTED** [review]

File: .github/workflows/ci-build.yml (hunk 2 above)
Comment (sawood14012) [reply .github/workflows/ci-build.yml]: done

File: qa/actions-run-tests.sh (hunk 1 above)
Comment (arajkumar) [reply qa/actions-run-tests.sh:1]: @sawood14012 , You can do a similar workaround done for backbone? i.e install virtualenv package.

Comment (sawood14012) [reply qa/actions-run-tests.sh:1]: yup will do

**Review by arajkumar — CHANGES_REQUESTED** [review]
Can you add coverage and build status badge in README.md?

[codecov](https://codecov.io/gh/fabric8-analytics/f8a-stacks-report)

CI


Also it would be great if you can add a step to validate the openshift/template.yaml also.

**Comment by sawood14012** [comment]
> Can you add coverage and build status badge in README.md?
> 
> Also it would be great if you can add a step to validate the openshift/template.yaml also.

okay will do

File: .github/workflows/ci-build.yml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    - name: Install oc
+      uses: redhat-actions/oc-installer@v1
+      with:
+        version: '4.6'
+    - name: Run oc process
+      run: oc process --local -f openshift/template.yaml -o yaml > blueprint.yaml
+    - name: validate openshift yaml
+      uses: instrumenta/kubeval-action@master
+      with:
+          files: blueprint.yaml
```
Comment (arajkumar) [inline .github/workflows/ci-build.yml:55]: You may need to run this as a separate job.

File: .github/workflows/ci-build.yml [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+      uses: codecov/codecov-action@v1
+      with:
+        token: ${{ secrets.CODECOV_TOKEN }}
+        file: ./coverage.xml
+    - name: Install oc
+      uses: redhat-actions/oc-installer@v1
+      with:
+        version: '4.6'
+    - name: Run oc process
+      run: oc process --local -f openshift/template.yaml -o yaml > blueprint.yaml
```
Comment (arajkumar) [inline .github/workflows/ci-build.yml:51]: How this works when the template needs mandatory param?

File: .github/workflows/ci-build.yml (hunk 3 above)
Comment (sawood14012) [reply .github/workflows/ci-build.yml:55]: Yup I will make it as a different job

File: .github/workflows/ci-build.yml (hunk 4 above)
Comment (sawood14012) [reply .github/workflows/ci-build.yml:51]: https://github.com/instrumenta/kubeval/issues/33 I was just trying out a solution , can you please tell me what mandatory param ?

Comment (sawood14012) [reply .github/workflows/ci-build.yml:51]: The solution works : https://github.com/sawood14012/f8a-stacks-report/runs/1565527998?check_suite_focus=true

Comment (arajkumar) [reply .github/workflows/ci-build.yml:51]: Mandatory param is one which should to passed while applying template. i.e using -p to `oc process`.

File: .github/workflows/ci-build.yml (hunk 3 above)
Comment (arajkumar) [reply .github/workflows/ci-build.yml:55]: @sawood14012 , Also you need to create another job which just builds docker image to ensure Dockerfile sanity.

File: .github/workflows/ci-build.yml (hunk 4 above)
Comment (sawood14012) [reply .github/workflows/ci-build.yml:51]: process command just provides us list of objects right , anyways we can add and override parameters , i found this in the documentation 
"You can override any parameter values defined in the file by adding the -p option for each <name>=<value> pair you want to override. A parameter reference may appear in any text field inside the template items." 
https://docs.openshift.com/container-platform/3.6/dev_guide/templates.html

Comment (sawood14012) [reply .github/workflows/ci-build.yml:51]: sorry for late replies i was on a leave today

File: .github/workflows/ci-build.yml (hunk 3 above)
Comment (sawood14012) [reply .github/workflows/ci-build.yml:55]: done

File: .github/workflows/ci-build.yml (hunk 4 above)
Comment (arajkumar) [reply .github/workflows/ci-build.yml:55]: We can merge this once you convert the yaml validation as separate job.

**Review by arajkumar — CHANGES_REQUESTED** [review]

File: .github/workflows/ci-build.yml [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    - name: Install virtualenv
+      run: pip install -U pip && pip install virtualenv
+    - name: Run Unit Tests
+      run: bash qa/runtests.sh
+    - name: Upload coverage to Codecov
+      uses: codecov/codecov-action@v1
+      with:
+        token: ${{ secrets.CODECOV_TOKEN }}
+        file: ./coverage.xml
+  openshift-yaml-validate:
```
Comment (sawood14012) [inline .github/workflows/ci-build.yml:46]: @arajkumar i have did it as a seperate job

Comment (sawood14012) [reply .github/workflows/ci-build.yml:46]: let me add badges and update

**Review by arajkumar — APPROVED** [review]
lgtm

**Pull request merged**